pkg/operation/botanist/waiter.go

```diff
@@ -84,7 +84,7 @@ func (b *Botanist) WaitUntilEtcdReady(ctx context.Context) error {
 				lastErrors = multierror.Append(lastErrors, fmt.Errorf("%s reconciliation in process", etcd.Name))
 			case etcd.Status.LastError != nil:
 				lastErrors = multierror.Append(lastErrors, fmt.Errorf("%s reconciliation errored with %s", etcd.Name, err))
-			case !etcd.Status.Ready:
+			case etcd.Status.Ready == nil || !*etcd.Status.Ready:
```

Comment:
Nit

```suggestion
			case !utils.IsTrue(etcd.Status.Ready):
```

with `"github.com/gardener/gardener/pkg/utils"`